### DISCUSSION TITLE
bootstrap 5 cant import

### DISCUSSION THREAD

**Discussion by ntd0609** [twbs/bootstrap#33570] [General]
ERROR in node_modules/bootstrap/dist/js/bootstrap.esm.js
Module parse failed: Unexpected token (1148:15)
You may need an appropriate loader to handle this file type.
| 
|   _getConfig(config) {
|     config = { ...Default$8,
|       ...config
|     };
 @/node_modules/babel-loader/lib??ref--2-0!/node_modules/vue-loader/lib??vue-loader-options!./List.vue?vue&type=script&lang=js& 35:0-34

----------
at line 35 I import bootstrap 5
import bootstrap from "bootstrap";
Please help me

**Comment by kiehnek** [comment]
Can you share a minimal demo? (for example with [codesandbox](https://codesandbox.io/)).

**Comment by iyulan** [comment]
same error with beta3. Use beta1 or beta2

**Reply by ElmmmQin** [reply]
The only useful answer, thannnnnnnnnks!

**Comment by kiehnek** [comment]
What are you importing (`js`, `css`, `scss`)? What's your bundler? How is your bundler setup?

**Comment by xdvarpunen** [comment]
There is no export default in .esm file. This is what it exports:

```js
export { Alert, Button, Carousel, Collapse, Dropdown, Modal, Offcanvas, Popover, ScrollSpy, Tab, Toast, Tooltip };
```

To import bootstrap use following example:

```html
<script>
import HelloWorld from './components/HelloWorld.vue'
// import { Alert } from 'bootstrap/dist/js/bootstrap.esm'
import * as name from 'bootstrap/dist/js/bootstrap.esm'

export default {
  name: 'App',
  components: {
    HelloWorld
  },
  mounted(){
    // console.log(Alert);
    console.log(name.Alert)
  }
}
</script>
```

**Reply by khassepaz** [reply]
you are my hero :))

**Reply by kiingsway** [reply]
The `import * as bootstrap from 'bootstrap';` worked as well when trying to import the JS from NavBar. Thanks!!

**Comment by ntd0609** [comment]
@korki43 
When I import another component 
Ex : Alert -  it's run bundle right way
But if I import all as document
import bootstrap from "bootstrap"
Opp... err will display
I tested err same when I import component that require depend on poopver 
-------
My bundle I use webpack

**Comment by kiehnek** [comment]
@ntd0609 see the example above by @xdvarpunen. Bootstrap doesn't provide a default export. You can import specific components using `import { Alert, Collapse } from "bootstrap"`. If you want to import all, you can use `import * as bootstrap from "bootstrap"`.

**Comment by ntd0609** [comment]
@korki43 
I will try your method you point by use import * as bootstrap from "bootstrap" and will respond asap result
But I remind that I only read document maybe conflict document 
![Screenshot_2021-04-04-07-55-33-52](https://user-images.githubusercontent.com/9030478/113495417-c2d00380-951b-11eb-9792-5730e3ff4da3.jpg)

**Comment by Katrienvh** [comment]
I got the same error using ES6 and webpack. My loader in `webpack.config.js` excluded the `/node_modules/`. It caused that the bootstrap js was not getting compiled.

My solution

`import * as bootstrap from "bootstrap/dist/js/bootstrap.js"`

added an extra webpack loader that handled the bootstrap javascript without excluding the node_modules.

```
      {
        test: /bootstrap\.js$/,
        loader: 'babel-loader',
        query: {
          cacheDirectory: true,
          presets: ['@babel/preset-env']
        }
      }
